controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -795,6 +797,80 @@ func TestRebootWithSuffixedAnnotation(t *testing.T) {
 	)
 }
 
+// TestRebootWithServicing tests full reboot cycle with suffixless
+// annotation and servicing.
+func TestRebootWithServicing(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = make(map[string]string)
+	host.Annotations[metal3api.RebootAnnotationPrefix] = ""
+	host.Status.PoweredOn = true
+	host.Status.Provisioning.State = metal3api.StateProvisioned
+	host.Spec.Online = true
+	host.Spec.Image = &metal3api.Image{URL: "foo", Checksum: "123"}
+	host.Spec.Image.URL = "foo"
+	host.Spec.Firmware = &metal3api.FirmwareConfig{
```

Comment:
Looking at the recent version of checkServicing, is this field even handled any more? Remember: there are two firmware settings API, I think you're only handling HFS now?

Reply:
Good pickup. Does this mean I need to add an HFS CR mockup as well?

Reply:
I tried a couple things but still without luck - if you have time I'd appreciate pointers. I tried different things (some still present but commented out) but I'm still failing going over 25 iterations.

Reply:
You don't have this part https://github.com/metal3-io/baremetal-operator/pull/1689/files#diff-a69d9029388ab766ed36b32180145f52785a9d4a153775510dbddfa928a72e1cR1344-R1347 from my PoC patch. Without it, setting `Spec.Firmware` does nothing regardless of servicing and HUP.

Reply:
(and you probably need another test for HFS case)

Reply:
w/r/t to missing parts - that was a massive stuff-up and misunderstanding. Once I dropped that commit, with the short assignment fix and with the understanding of old firmware settings API I finally managed to get things to pass locally. Let's see what CI thinks.